packages/synapse-react-client/src/components/QueryWrapper/useTableRowSelection.ts

```diff
@@ -18,6 +20,49 @@ type UseTableRowSelectionReturn = {
   selectedRows: Row[]
   setSelectedRows: React.Dispatch<React.SetStateAction<Row[]>>
   hasSelectedRows: boolean
+  getIsRowSelected: (row: Row) => boolean
+}
+
+function getRowSelectionEqualityComparator(
+  row: Row,
+  columnModels?: ColumnModel[],
+  rowSelectionPrimaryKey?: string[],
+): (r1: Row, r2: Row) => boolean {
+  // Worst-case scenario, use deep equality to compare rows
+  // This will fail from the user's perspective if any data changes in the rows since the last time the selection was updated.
+  let comparator: (r1: Row, r2: Row) => boolean = isEqual
+  if (rowSelectionPrimaryKey && columnModels) {
+    // If the `rowSelectionPrimaryKey` is defined, a row is selected if we have a PK match
+    // Even this selection is not guaranteed to persist if data changes, e.g. if column order changes and the PK now has a different column index
+    comparator = (r1: Row, r2: Row) => {
+      const r1PrimaryKeyValues = rowSelectionPrimaryKey.map(
+        key => r1.values[columnModels.findIndex(cm => cm.name === key)]!,
+      )
+      const r2PrimaryKeyValues = rowSelectionPrimaryKey.map(
+        key => r2.values[columnModels.findIndex(cm => cm.name === key)]!,
+      )
+      return isEqual(r1PrimaryKeyValues, r2PrimaryKeyValues)
+    }
+  } else if (row.rowId) {
+    // If there's a rowId, we can use that as a fallback
+    comparator = (r1: Row, r2: Row) => r1.rowId === r2.rowId
+  }
+
+  return comparator
+}
```

Comment:
I'm not sure if we've committed to persisting selection between sessions, but that was a consideration when deciding to use primary key instead of just naively using a deep equality check.

Reply:
Thank for falling back to the rowId as well.  Since CRC no longer supports joining the study, I think the CRC Researcher portal will no longer be used.  But I think it would still work properly with this logic